File: src/main/java/com/github/analuciabolico/arduinomakerbff/entity/Block.java
package com.github.analuciabolico.arduinomakerbff.entity;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Data
@Builder
@ToString
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "classes")
public class Block {
    @Id
    private String id;
    private String name;
    private String code;
    private String include;
    private String type;
    private Integer quantity;
}

File: src/main/kotlin/com/github/analuciabolico/arduinomakerbff/entity/Block.kt
package com.github.analuciabolico.arduinomakerbff.entity

import com.github.analuciabolico.arduinomakerbff.dto.BlockResponseDto
import org.bson.types.ObjectId
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document


@Document(collection = "blocks")
class Block(
    @Id
    private val id: ObjectId? = ObjectId(),
    private val name: String? = null,
    private val code: String? = null,
    private val include: String? = null,
    private val type: BlockTypesEnum? = null,
    private val board: String? = null,
    private val quantity: Int? = null,
) {
    fun toDto(): BlockResponseDto {
        return BlockResponseDto(
            id = this.id.toString(),
            name = this.name,
            code = this.code,
            include = this.include,
            type = this.type,
            board = this.board,
            quantity = this.quantity,
        )
    }
}